tools/docker/docker-compose-stage.yaml

```diff
@@ -84,6 +84,9 @@ services:
       start_period: 5s
   eda-api:
     image: ${EDA_IMAGE:-quay.io/ansible/eda-server}:${EDA_IMAGE_VERSION:-main}
+    build:
```

Comment:
@AlanCoding following up my [last comment](https://github.com/ansible/eda-server/pull/978#pullrequestreview-2191159938), one of the purpose of the -stage is to run the application with prebuilt images, if you want to build the image from the source code, you have the -dev image. This change would also break some CI pipelines. I think this file should not build the image and it is not necessary because you could override the variable to use your local one if you want that. 

What we can do, is to remove `EDA_IMAGE_VERSION` and use just `EDA_IMAGE` for the whole url as we already do in the -dev image.

Reply:
> What we can do, is to remove EDA_IMAGE_VERSION and use just EDA_IMAGE for the whole url as we already do in the -dev image.

I did find that type-inconsistent. I agree, if I'm expecting to be able to swap the dockerfile, then I need to be able to give the image in the same format. We could use a separate variable for the full image, like `EDA_IMAGE=$(EDA_IMAGE_BASE):$(EDA_IMAGE_VERSION)`. But I will just make the change like you suggest.

https://github.com/ansible/eda-server/blob/726a3a9ddff4c7d0336da04e7fd5e2eb96be7e80/tools/docker/docker-compose-dev.yaml#L114

This will lead to a little more legwork of building images with the dev dockerfile and then running them with -stage, but that is fine, given that it's intended.

Reply:
I think that replicating what we do in -dev file is enough.
`image: ${EDA_IMAGE:-quay.io/ansible/eda-server:main}`

Reply:
On second look, this is too big of a change for me to changing the meaning of `EDA_IMAGE` because of how it is referenced in the Makefile. There is an existing `EDA_IMAGE_TAG` which would indicate the image without the version tag, but the contract right now doesn't use that variable.